### DISCUSSION TITLE
How to add modal popup with custom HTML/layout?

### DISCUSSION THREAD

**Discussion by code-by** [telegraf/telegraf#1748] [Q&A] [answered]
I'd like to add popup/modal page with my custom HTML, ex
https://drive.google.com/file/d/1GqCe0mUBFqlSbQgeW5H07TR6feovDqys/view?usp=share_link
is this possible do with telegraf?
thank you

**Comment by MKRhere** [comment, marked as answer]
This is a Telegram Web App [(read docs)](https://core.telegram.org/bots/webapps). In Telegraf, you simply send an button with `Markup.button.webApp(text, url)`. On clicking the button, the web app opens in a webview.
